src/main/java/io/stargate/sgv2/jsonapi/service/shredding/NamedValueContainer.java

```diff
@@ -0,0 +1,56 @@
+package io.stargate.sgv2.jsonapi.service.shredding;
+
+import io.stargate.sgv2.jsonapi.util.PrettyPrintable;
+import io.stargate.sgv2.jsonapi.util.PrettyToStringBuilder;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Base for all containers that hold {@link NamedValue}s.
+ *
+ * <p>Useful so we can deal with a container of {@link NamedValue}s in a generic way, provides some
+ * helper methods, and implements {@link PrettyPrintable}.
+ *
+ * @param <NameT> The type of the name, this is the key in the map
+ * @param <ValueT> The type of the value stored in the {@link NamedValue}
+ * @param <NvT> The type of the {@link NamedValue} stored in the map
```

Comment:
This seems redundant... ideally wouldn't we just need `NameT` and `ValueT`, `NvT` being `NamedValue<NameT, ValueT>`? Type parameter won't help creating instances anyway.